src/RestaurantDetailDrawer.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import Drawer from './common/Drawer.tsx';
+import { Restaurant } from './util/type.js';
+import { CATEGORY_IMAGES, NO_EXIST_RESTAURANT } from './util/constant.ts';
+
+type RestaurantDetailDrawerProps = {
+  isOpenDrawer: boolean;
+  restaurantId: number;
+  onToggleDrawer: (id?: number) => void;
+};
+
+type RestaurantDetailDrawerState = {
+  restaurant: Omit<Restaurant, "id">;
+}
+
+class RestaurantDetailDrawer extends React.Component<
+  RestaurantDetailDrawerProps,
+  RestaurantDetailDrawerState
+> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      restaurant: this.fetchRestaurantById()
+    };
+  }
+
+  componentDidUpdate(prevProps:RestaurantDetailDrawerProps): void {
+    if (this.props.restaurantId !== prevProps.restaurantId) {
+      this.setState({
+        restaurant: this.fetchRestaurantById()
+      });
+    }
+  }
+
+  fetchRestaurantById() {
+    const rawRestaurantList = localStorage.getItem('restaurantList');
+    if (!rawRestaurantList) return NO_EXIST_RESTAURANT;
+    const restaurantList = JSON.parse(rawRestaurantList);
+    return restaurantList.find((restaurant:Restaurant) => +restaurant.id === +this.props.restaurantId) ?? NO_EXIST_RESTAURANT;
+  }
+
+  render() {
+    return (
+      <Drawer isOpenDrawer={this.props.isOpenDrawer}>
+        <div className='restaurant__category'>
+          <img src={CATEGORY_IMAGES[this.state.restaurant.category]} alt={this.state.restaurant.category} className="category-icon" />
+        </div>
+        <h3 className="restaurant__name text-subtitle">
+          {this.state.restaurant.title}
+        </h3>
+        <span className="restaurant__distance text-body">
+          캠퍼스로부터 {this.state.restaurant.distance}분 내
+        </span>
+        <p className="text-body">
+          {this.state.restaurant.description}
+        </p>
+        <p className="restaurant__link text-body">
+          {this.state.restaurant.link ?? ''}
+        </p>
+        <button
+          type="button"
+          className="button button--secondary text-caption"
+          onClick={() => this.props.onToggleDrawer()}
+        >
```

Comment:
화살표 함수는 컴포넌트 렌더링마다 새로 생성되므로, 메서드 / 변수로 분리하거나... 인자가 동일하게 정해져있다면 아래와 같이 사용해주세요!

```suggestion
        <button
          type="button"
          className="button button--secondary text-caption"
          onClick={this.props.onToggleDrawer}
        >
```